src/docs/introduction/upgrading-to-stencil-three.md

```diff
@@ -0,0 +1,295 @@
+| `onCompositionUpdateCapture` | `onCompositionupdateCapture` |
+
+## Output Targets
+
+### `dist-custom-elements` Output Target
+#### Add `customElementsExportBehavior` to Control Export Behavior
+`customElementsExportBehavior` is a new configuration option for the output target.
+It allows users to configure the export behavior of components that are compiled using the output target.
+By default, this output target will behave exactly as it did in Stencil v2.0.0.
+
+This config option provides additional behaviors that will alter the default component export OR custom element definition behaviors
+for this target. The desired behavior can be set via the following in a project's Stencil config:
+
+```ts
+// stencil.config.ts
+import { Config } from '@stencil/core';
+export const config: Config = {
+  outputTargets: [
+    {
+      type: 'dist-custom-elements',
+      customElementsExportBehavior: 'default' | 'single-export-module',
+    },
+    // ...
+  ],
+  // ...
+};
+```
+
+| Option                 | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
+| ---------------------- |-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `default`              | No additional re-export or auto-definition behavior will be performed.<br><br>This value will be used if no explicit value is set in the config, or if a given value is not a valid option.                                                                                                                                                                                                                                                                                 |
+| `single-export-module` | All component and custom element definition helper functions will be exported from the `index.js` file in the output directory (see [Defining Exported Custom Elements](/docs/custom-elements#defining-exported-custom-elements) for more information on this file's purpose). This file can be used as the root module when distributing your component library, see [Distributing Custom Elements](/docs/custom-elements#distributing-custom-elements) for more details. |
```

Comment:
We'll need to update this to account for the missing options too:
```suggestion
| `default`                     | No additional re-export or auto-definition behavior will be performed.<br><br>This value will be used if no explicit value is set in the config, or if a given value is not a valid option.                                                                                                                                                                                                               |
| `auto-define-custom-elements` | A component and its children will be automatically defined with the `CustomElementRegistry` when the component's module is imported.                                                                                                                                                                                                                                                                      |
| `bundle`                      | A utility `defineCustomElements()` function is exported from the `index.js` file of the output directory. This function can be used to quickly define all Stencil components in a project on the custom elements registry.                                                                                                                                                                                 |
| `single-export-module`        | All component and custom element definition helper functions will be exported from the `index.js` file in the output directory (see [Defining Exported Custom Elements](#defining-exported-custom-elements) for more information on this file's purpose). This file can be used as the root module when distributing your component library, see [below](#distributing-custom-elements) for more details. |
```

Reply:
Or were you omitting those options because they're covered separately below?

Reply:
Nope, better they all be here for now